omod/src/main/java/org/openmrs/module/fhir/providers/RestfulEncounterResourceProvider.java

```diff
@@ -79,7 +79,7 @@ public Encounter getResourceById(@IdParam IdType theId) {
 	 */
 	@Search
 	public List<Encounter> findEncountersByPatientIdentifier(
-			@RequiredParam(name = Patient.SP_IDENTIFIER) ReferenceParam identifier) {
+			@RequiredParam(name = Encounter.SP_PATIENT + "." + Patient.SP_IDENTIFIER) ReferenceParam identifier) {
```

Comment:
The endpoint for getting observations for a patient doesn't manually insert a period. Instead, it uses a thing called `chainWhitelist` to accomplish this. I think we should update this PR to use `chainWhitelist`. See
[example in observations resource](https://github.com/openmrs/openmrs-module-fhir/blob/master/omod/src/main/java/org/openmrs/module/fhir/providers/RestfulObservationResourceProvider.java#L139-L140) for an example of this. Also see [here](https://github.com/jamesagnew/hapi-fhir/blob/dff2fdd3cfb07fcb6e64826e17c37f87768b63f7/hapi-fhir-base/src/main/java/ca/uhn/fhir/rest/annotation/OptionalParam.java#L71) and [here](http://hapifhir.io/apidocs/ca/uhn/fhir/rest/annotation/RequiredParam.html#chainWhitelist--) for some documentation in hapi-fhir regarding `chainWhitelist`


```suggestion
			@RequiredParam(name = Encounter.SP_PATIENT, chainWhitelist = { Patient.SP_IDENTIFIER }) ReferenceParam identifier) {
```